tests/simulation/sim_test.go

```diff
@@ -211,3 +239,287 @@ func TestFullAppSimulation(t *testing.T) {
+		config,
+		SimDBBackend,
+		SimDBName,
+		simutils.FlagVerboseValue,
+		simutils.FlagEnabledValue,
+	)
+	if skip {
+		t.Skip("skipping application simulation")
+	}
+	require.NoError(t, err, "simulation setup failed")
+
+	defer func() {
+		if err := db.Close(); err != nil {
+			t.Errorf("Error closing new database: %v", err)
+		}
+		if err := os.RemoveAll(dir); err != nil {
+			t.Errorf("Error removing directory %s: %v", dir, err)
+		}
+	}()
+
+	appOptions := make(cosmossimutils.AppOptionsMap, 0)
+	appOptions[server.FlagInvCheckPeriod] = simutils.FlagPeriodValue
+	appOptions[flags.FlagHome] = dir
+	simApp, err := simutils.NewSimApp(logger, db, appOptions, interBlockCacheOpt(), baseapp.SetChainID(SimAppChainID))
+	require.NoError(t, err)
+
+	// Run randomized simulation
+	blockedAddresses := simApp.ModuleAccountAddrs()
+	_, simParams, simErr := simulation.SimulateFromSeed(
+		t,
+		os.Stdout,
+		simApp.BaseApp,
+		simutils.AppStateFn(
+			simApp.AppCodec(),
+			simApp.SimulationManager(),
+			simApp.BasicManager().DefaultGenesis(simApp.AppCodec()),
+		),
+		cosmossim.RandomAccounts,
+		cosmossimutils.SimulationOperations(simApp, simApp.AppCodec(), config),
+		blockedAddresses,
+		config,
+		simApp.AppCodec(),
+	)
+	require.NoError(t, simErr)
+
+	err = simutils.CheckExportSimulation(simApp, config, simParams)
+	require.NoError(t, err)
+
+	simutils.PrintStats(db)
+
+	t.Log("exporting genesis")
+	// export state and simParams
+	exported, err := simApp.ExportAppStateAndValidators(false, []string{}, []string{})
+	require.NoError(t, err)
+
+	t.Log("importing genesis")
+	newDB, newDir, _, _, err := cosmossimutils.SetupSimulation(
+		config,
+		SimDBBackend+"_new",
+		SimDBName+"_new",
+		simutils.FlagVerboseValue,
+		simutils.FlagEnabledValue,
+	)
+
+	require.NoError(t, err, "simulation setup failed")
+
+	defer func() {
+		if err := newDB.Close(); err != nil {
+			t.Errorf("Error closing new database: %v", err)
+		}
+		if err := os.RemoveAll(newDir); err != nil {
+			t.Errorf("Error removing directory %s: %v", newDir, err)
+		}
+	}()
+
+	newSimApp, err := simutils.NewSimApp(
+		logger,
+		newDB,
+		appOptions,
+		interBlockCacheOpt(),
+		baseapp.SetChainID(SimAppChainID),
+	)
+	require.NoError(t, err)
+
+	var genesisState app.GenesisState
+	err = json.Unmarshal(exported.AppState, &genesisState)
+	require.NoError(t, err)
+
+	defer func() {
+		if r := recover(); r != nil {
+			err := fmt.Sprintf("%v", r)
+			if !strings.Contains(err, "validator set is empty after InitGenesis") {
+				panic(r)
+			}
+			logger.Info("Skipping simulation as all validators have been unbonded")
+			logger.Info("err", err, "stacktrace", string(debug.Stack()))
+		}
+	}()
+
+	ctxA := simApp.NewContext(true, tmproto.Header{
+		Height:  simApp.LastBlockHeight(),
+		ChainID: SimAppChainID,
+	}).WithChainID(SimAppChainID)
+	ctxB := newSimApp.NewContext(true, tmproto.Header{
+		Height:  simApp.LastBlockHeight(),
+		ChainID: SimAppChainID,
+	}).WithChainID(SimAppChainID)
```

Comment:
What's the difference between two context? I think we can be more precise in the variable name